tests/test_fmi3.py

```diff
@@ -283,6 +301,88 @@ def test_invalid_log_level(self, log_level):
         with pytest.raises(FMUException, match = msg):
             FMUModelME3(fmu_path, log_level = log_level, _connect_dll = False)
 
+    @pytest.mark.parametrize("variable_name, value, expected_dtype",
+        [
+            ("Float64_continuous_input", 3.14, np.double),
+            ("Float32_continuous_input", np.float32(3.14), np.float32),
```

Comment:
Note: `np.float32` for the input value is required since it does not have an exact float representation.